butane_cli/src/lib.rs

```diff
@@ -259,10 +261,46 @@ pub fn get_migrations() -> Result<FsMigrations> {
     Ok(migrations::from_root(root))
 }
 
+/// The members of a workspace are given a PackageId, which contains the directory
+/// name however it needs to be extracted.
+pub fn extract_package_directory(
+    package_id: cargo_metadata::PackageId,
+) -> Result<std::path::PathBuf> {
+    let mut file_url = package_id
+        .repr
+        .split_once(&"path+".to_string())
+        .unwrap()
+        .1
+        .to_string();
+    file_url.truncate(file_url.len() - 1);
+    let url = Url::parse(file_url.as_str()).expect("Failed to parse URL");
+    let file_path = url.to_file_path().unwrap();
+    Ok(file_path)
+}
+
+/// Find a .butane directory to act as the base for butane.
 pub fn base_dir() -> Result<PathBuf> {
-    std::env::current_dir()
-        .map(|d| d.join(".butane"))
-        .map_err(|e| e.into())
+    let current_directory = std::env::current_dir()?;
+    let local_butane_dir = current_directory.join(".butane/");
+
+    if !local_butane_dir.exists() {
+        let metadata = MetadataCommand::new().no_deps().exec()?;
+        let workspace_members = metadata.workspace_members;
+
+        // Find the first workspace member with a .butane
```

Comment:
I'm a bit uneasy about just picking the first one without any further interaction. For one thing I don't think it's even stable. I tried running it from a workspace containing a couple different projects both with butane set up and while I mostly got the same project, sometimes I got a different one.

I agree it's a nice quality-of-life improvement to be able to run `butane` from the workspace root, and if there's only one workspace project with a `.butane` directory it makes sense to pick that one. But I think we need to check whether there's more than one, and not proceed without confirmation if there are choices. We can offer a flag (like cargo's `-p`) to select the project to operate on.

If we're in a subdirectory of a project it probably makes sense to automatically select that project since the working directory is giving us a hint about what the user intended.